File: Microservicios/rack-service/src/main/java/com/tutorial/rack_service/service/RackService.java
package com.tutorial.rack_service.service;

import com.tutorial.rack_service.entity.Rack;
import com.tutorial.rack_service.model.Reservation;
import com.tutorial.rack_service.repository.RackRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

import java.util.List;

@Service
public class RackService {

    @Autowired
    RackRepository rackRepository;

    public List<Rack> getAll() {
        return rackRepository.findAll();
    }

    public List<Rack> getRacksByYearAndMonth(String year, String month) {
        return rackRepository.findRacksByYearAndMonth(year, month);
    }

    public Rack getRackById(int id) {
        return rackRepository.findById(id).orElse(null);
    }

    public Rack save(Rack rack) {
        return rackRepository.save(rack);
    }

    public Rack update(Rack rack, int reservationId) {
        rack.getReservations().add(reservationId);
        return rackRepository.save(rack);
    }
}

File: Microservicios/front-end/src/services/rack.service.js
import httpClient from '../http-common';

const getAll = () => {
  return httpClient.get('/rack-service/api/v1/racks');
}

const getById = (id) => {
  return httpClient.get(`/rack-service/api/v1/racks/${id}`);
}

const getByYearMonth = (year, month) => {
  return httpClient.get(`/rack-service/api/v1/racks/year/${year}/month/${month}`);
}

const create = (data) => {
  return httpClient.post('/rack-service/api/v1/racks', data);
}

const update = (id, data) => {
  return httpClient.put(`/rack-service/api/v1/racks/${id}`, data);
}

export default { getAll, getById, getByYearMonth, create, update };